app/src/main/java/com/example/Egida/presentation/viewModel/MainViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.Egida.presentation.viewModel
+
+import android.text.TextUtils
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.Egida.Dependencies
+import com.example.Egida.domain.entity.User
+import com.example.Egida.domain.useCase.UserUseCase
+import kotlinx.coroutines.launch
+
+
+class MainViewModel(
+
+) : ViewModel() {
+
+    private val userUseCase: UserUseCase by lazy { Dependencies.getUserUseCase() }
+    var email: String = ""
+    var password: String = ""
+    var doublePassword: String = ""
+    private val user = User(email, password)
+    var toast = MutableLiveData<String>()
+
+    fun checkingUser(): Boolean {
+        return userUseCase.checkingUser()
+    }
+
+    fun addUser() {
+        if (!TextUtils.isEmpty(email)
+            && !TextUtils.isEmpty(password)
+            && !TextUtils.isEmpty(doublePassword)
+        ) {
+            if (password == doublePassword) {
+                viewModelScope.launch {
+                    userUseCase.addUser(user)
+                    toast.value = "Проверьте вашу почту для подтверждения регистрации"
```

Comment:
Better ways:
1. Replace string with R.string.error_id
2. Even better way for common errors: sealed class with a separate name, witch will be converted to string at UI.